pkg/module/sync/runner.go

```diff
@@ -180,17 +188,21 @@ func (r *Runner) updateAvailableManifestSpec(ctx context.Context, manifestObj *v
 }
 
 func NeedToUpdate(manifestInCluster, manifestObj *v1beta2.Manifest, moduleStatus *v1beta2.ModuleStatus,
-	moduleTemplateGeneration int64,
+	module *common.Module,
 ) bool {
+	if manifestInCluster == nil && module.IsUnmanaged {
+		return false
+	}
 	if manifestInCluster == nil || moduleStatus == nil { // moduleStatus is nil in case of mandatory module
 		return true
 	}
-	if moduleStatus.Template != nil && moduleStatus.Template.GetGeneration() != moduleTemplateGeneration {
+	if moduleStatus.Template != nil && moduleStatus.Template.GetGeneration() != module.Template.GetGeneration() {
 		return true
 	}
 	return manifestObj.Spec.Version != moduleStatus.Version ||
 		manifestObj.Labels[shared.ChannelLabel] != moduleStatus.Channel ||
-		moduleStatus.State != manifestInCluster.Status.State
+		moduleStatus.State != manifestInCluster.Status.State ||
+		module.IsUnmanaged != manifestInCluster.IsUnmanaged()
```

Comment:
I think unmanaged condition need to be treated separately, if `module.IsUnmanaged`, we only update manifest cr if it's not contains `UnmanagedAnnotation`, but doesn't update if other conditions are different.

Reply:
Can you check again? Current logic should be sufficient in my opinion?